src/Simplify_Mul.cpp

```diff
@@ -69,6 +69,8 @@ Expr Simplify::visit(const Mul *op, ExprInfo *bounds) {
 
         if (rewrite((x + c0) * c1, x * c1 + fold(c0 * c1), !overflows(c0 * c1)) ||
             rewrite((c0 - x) * c1, x * fold(-c1) + fold(c0 * c1), !overflows(c0 * c1)) ||
+            rewrite((0 - x) * y, 0 - x * y) ||
+            rewrite(x * (0 - y), 0 - x * y) ||
```

Comment:
Do these rules obey the reduction order? @abadams @jn80842

Reply:
Yes, promoting an add or sub to the root is one of the ordering terms, and the rule is a tie for all of the ordering terms before it.